### ISSUE TITLE
Reconcile partial in Bank Statement

### ISSUE THREAD

**Issue by luistorresm** [odoo/odoo#4466]
1) Steps to reproduce the issue you have observed
First you create 2 Invoices to the same customer, and validate.
After create a new bank statement, to the customer invoices, & for an amount less that the sum of the two invoices.
You reconcile the bank statement.

Can find a demonstrative video here: 
http://youtu.be/QeIcnsrqLrg

2) The result you observed
I can't indicate that amount is for each invoice

 3) The result you expected
I could indicate the amount to each invoice

**Comment by Whisno** [comment]
Ok, so after discussion, this is not a use case that we want to support through the bank reconciliation interface.
It can be done indirectly by creating a write-off (possibly several to split the transaction) and then using the manual reconciliation interface. This is not especially straightforward but regarding the fact that this use case is very uncommon, we'd rather keep the software simple.

**Comment by nhomar** [comment]
Hello.

Due to the new accounting planning as you can read, Fabien comment about Account Voucher which will be killed in favor of new bank statement:

https://docs.google.com/document/d/1MahXh0TdfjI0ohdTSz_c8cML7HqxvcAtvSAY63N20Tc/edit

then, said this we must have all possibilities availables throught account voucher on bank statement, and viceversa (even if functionally they are pretty different, even I am not agreed with kill account voucher it is a dection taken by Odoo).

Then we have now some bugs which block this possibility thorught bank statement ("process proposed)" and answer "we decided not support such feature" is incorrect IMHO due to on business process I can not change or remove one step of 2 click per one of 20 when our customers come from older versions, I think it is not correct.

I can do this on Account voucher since version 6.0 << it is a fact.

Then talking about less generic stuff and talking about this use case:

When this feture is necesary?

Imagine you have with your customer/supplier different credt/Early Payment discount criteria, Example:

(I will imagine it is the same for Sales and Purchase I will talk about sales now)

1.- Purchases over 1000€ 5% of discount for early payment.
2.- Purchases with 6 month of credit paid in 6 parts with 2% of early payment.
3.- Purchases with discounts higher than 10% I must give 10% if it is paid in advance and 5% if it is paid after deliver.

Do everything external currencies and the VAT is based on paid invoices not on invoices themself.

Then.... I can continue giving you cases but you can have several differents criterias of payments to different invices with differents of the total amounts due to differents criterias which are out of this scope.

Then actually in vouchers />> which "remember" will be a deprecated method.

I receive a payment and I just open the voucher and (after compute all cases, credit notes and discounts).

- Load a dollar payment <<external currency.
- Set the amount per invoice (let say he is paying 3 parts of case -2- plus one of them with early payment and 3 cases of case -3- and 10 cases of case -1- total 16 `payments with an amount plus its corresponding cradit notes).
- Validate it.
- Everything works fine (it has a bug now on v8.0 which avoid do that and remember it will be a deprecated feature) , system compute even the differencial exchange rate perfect.
- Go to bank statement and relate this payment to a line (feature not correctly working but AFAICK it will work perfect on v9, but feasible).

The process you mentioned me, is really too complex, imagine all this work which solve account voucher, DIfference with exchange rate, VAT paid, Reconciliation with credit Notes AND amount are never sending me wrong numbers on invoices (An remember that I can not do a GENERIC reconcile process because we have differents payments/discount criterias).

Then you offer us a cool BS interface which solve 70% of cases but the 30 more complicated simply I need to open aaaaaaaaallll accounting to the person who simply load payments in the system, is it really the approach you are offering me?.

Don't you think it is a little incorrect?

Dude. I don't want you cover 100% use cases, but the ones already worked on previous versions please mantain them or at least when we mention there is a regresion or a leak of things working "Considere them" please!!!!!

The request is fairly simple:

1.- Set specific amounts per invoices, It is not correct ask do the job manually because security resons, procedural reasons and future plans in terms of features that will be killed on 9.0.

When you are designing for companies with 1 payment per month, we (the ones which pay HUGE enterprises) are working with peoples with 20 payments per hour my friend **remember that**.

I hope it is more clear.

**Comment by qdp-odoo** [comment]
Hi guys,

indeed, we deliberately chose to disallow to encode the amount to reconcile on each selected invoice. The reasons are
* simply because we wanted to have a clean and simple widget. So we removed the optional features
* we also believe that the use case given in the bug description is uncommon. Most of the time when you receive a partial payment you want to assign totally to a given invoice (let's say the oldest)

If you think, that use case is vital for you, the given workaround is the following: 
1) on the reconcile interface in the bank statement, don't choose any invoice but rather create several lines each on the debtor account with the amounts to allocate respectively on each invoice (create a button if you do that often).
2) go in Periodical Processing \ Reconciliation \ Manual reconciliation and partially reconcile your invoices with their dedicated line coming from the bank

Now seriously speaking, i even find this easier than in v7.0 ! (you can't even argue with that: the user experience was awful! the red bullet isn't explicit, has to be clicked twice, the whole shit was flickering and no one knew they can modify the amount allocated on each invoice!)

Nhomar, your request has been considered, but we never guarantee to keep all the features as it when moving to newer versions. We made a simplification that aims to ease the process for most of people in most of the use cases, sorry if it's not *your* case, and moreover there exists a solution for those who really need this (read "there is no feature removed")... So that's the reasons this issue is not covered by any enterprise contract (even HUGE ones).

thanks for your comprehension,
Quentin

**Comment by nhomar** [comment]
@qdp-odoo 

I am not agreed at all, and I can understand you, but BTW, if you can not give me even the chance of the doubt not there are nothing to do here.

**Comment by siouf** [comment]
I agree that the use case of encoding the precise amount to partially reconcile more than one invoice is too specific. I also agree that it is better to keep the user interface as simple as possible (IMO Odoo's bank statement view is great because it is so simple!)

However, I do think that the use case of partial payments fully reconciling open invoices from the oldest to the newest and partially reconciling the latest invoice is not that much of a custom case.

As @qdp-odoo has said:
> Most of the time when you receive a partial payment you want to assign totally to a given invoice (let's say the oldest)

Agreed. But what if the partial payment is covering more than a single invoice? Then I would want to start assigning from the oldest, and then use the leftover to partially reconcile the last invoice. This use case seems fairly common to me.

What's great is that such functionality wouldn't mess up the user interface. No need for any additional button. The idea is simple. If multiple counterparts are selected, loop through them chronologically and partially reconcile the last invoice. Give the choice to confirm the partial reconciliation with the /!\ yellow signal next to the last invoice, or propose to create a write-off for full reconciliation of all selected invoices, including the last one.

In other words, treat the multiple invoices as if they were one big invoice. Seems like quite a lot more functionality at no cost in user interface simplicity!

**Comment by qdp-odoo** [comment]
indeed. I like this idea :+1: 

we definitively gonna do it in the future v9  :-)
